common/structs.go

```diff
@@ -100,12 +102,39 @@ type Equipment struct {
 	ReserveAmmo    int
 }
 
+type GrenadeProjectile struct {
+	EntityID int
+	Weapon   EquipmentElement
+	Thrower  *Player
+	Owner    *Player
+
+	// RandomID is used to distinguish different grenades (which potentially has the same, reused entityID) from each other.
+	randomID int64
+
+	CellOrigin
+}
+
+// CellOrigin is a position that is not given directly, but is computed from cell{X,Y,Z} and m_vecOrigin data properties.
+type CellOrigin struct {
+	Position r3.Vector
+}
+
 // Class returns the class of the equipment.
 // E.g. pistol, smg, heavy etc.
 func (e Equipment) Class() EquipmentClass {
 	return e.Weapon.Class()
 }
 
+// NewGrenadeProjectile creates a grenade projectile and sets.
+func NewGrenadeProjectile() *GrenadeProjectile {
```

Comment:
I think I'd prefer a sequence on the parser or something like that. No collision chances and faster.

It probably wouldn't ever be a problem either way but seems somewhat cleaner to me.

Reply:
Yeah I think I commented on the wrong line, my bad.
And yeah just leave it for now, should be fine.

One other thing tho: I think the ID should be renamed to UniqueID or something, otherwise it's implying the library consumer isn't supposed to use it.

Reply:
Good point w.r.t. your edited comment. I'll try and see if I can find a better name for it. `UniqueID` is not bad, but I'll think about it some more :)

Reply:
I moved that to a separate comment on the right line and yeah if you come up with something else that's good too.